packages/shortest/src/core/runner/index.ts

```diff
@@ -19,13 +22,29 @@ import {
 import { CacheEntry } from "../../types/cache";
 import { hashData } from "../../utils/crypto";
 import { TestCompiler } from "../compiler";
-import { TestReporter } from "./test-reporter";
 
-interface TestResult {
-  result: "pass" | "fail";
-  reason: string;
-  tokenUsage?: { input: number; output: number };
-}
+export const TokenMetricsSchema = z.object({
+  input: z.number().default(0),
+  output: z.number().default(0),
+});
+export type TokenMetrics = z.infer<typeof TokenMetricsSchema>;
+
+export const TestResultSchema = z.object({
+  test: z.any() as z.ZodType<TestFunction>,
+  status: z.enum(["pending", "running", "passed", "failed"]),
```

Comment:
Changed from `result` with `pass` & `fail` for consistency. There used to be 2 `TestResult` interfaces: here, and in `test-reporter.ts`.